File: 1889-CheckIfNumberIsASumOfPowersOfThree/1889-CheckIfNumberIsASumOfPowersOfThree.java
// Last updated: 7/31/2025, 6:04:26 PM
class Solution {
    public boolean checkPowersOfThree(int n) {
        while (n > 0) {
            if (n % 3 == 2) {
                return false;
            }
            n /= 3;
        }
        return true;
    }
}

File: 1889-CheckIfNumberIsASumOfPowersOfThree/1889-CheckIfNumberIsASumOfPowersOfThree.py
# Last updated: 7/31/2025, 6:04:21 PM
class Solution:
    def checkPowersOfThree(self, n: int) -> bool:
        while n > 0:
            if n % 3 == 2:
                return False
            n //= 3
        
        return True